src/main/scala/chisel3/util/Mux.scala

```diff
@@ -80,6 +80,17 @@ object MuxLookup {
 
     mappingx.foldLeft(defaultx) { case (d, (k, v)) => Mux(k === key, v, d) }
   }
+
+  /** Same as MuxLookup.apply but for EnumTypes
+    *
+    * @param key a key to search for
+    * @param default a default value if nothing is found
+    * @param mapping a sequence to search of keys and values
+    * @return the value found or the default if not
+    */
+  def fromEnum[S <: EnumType, T <: Data](key: S, default: T, mapping: Seq[(S, T)]): T = {
```

Comment:
what happens if you try to overload the `apply` function? Does the Scala compiler complain?

Reply:
I get a compile error `method apply is defined twice`. https://scastie.scala-lang.org/eO18f45oQgKIi2GiBhMNYg

Reply:
OK. Sad :(

Reply:
We're successfully overloading apply now 🙂 see https://github.com/chipsalliance/chisel/pull/3095